### PR TITLE
[ACA-4438] The document list gets reloaded when closing the document viewer

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

https://alfresco.atlassian.net/browse/ACA-4438

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: src/app/components/viewer/viewer.component.ts [hunk 1]
```diff
@@ -190,7 +190,6 @@ export class AppViewerComponent implements OnInit, OnDestroy {
   }
 
   onViewerVisibilityChanged() {
-    this.store.dispatch(new ReloadDocumentListAction());
```
Comment (DenysVuika) [inline src/app/components/viewer/viewer.component.ts:193]: causes
```
ERROR in src/app/components/viewer/viewer.component.ts:33:3 - error TS6133: 'ReloadDocumentListAction' is declared but its value is never read.
33   ReloadDocumentListAction,
     ~~~~~~~~~~~~~~~~~~~~~~~~
```

you probably need to cleanup the imports

Comment (sivakumar414ram) [reply src/app/components/viewer/viewer.component.ts:193]: Updated

**Review by DenysVuika — APPROVED** [review]

File: src/assets/app.extensions.json [hunk 2]
```diff
@@ -928,6 +928,9 @@
               "id": "app.viewer.toggleLock",
               "order": 100,
               "type": "custom",
+              "data": {
+                "actionType": "viewer"
```
Comment (DenysVuika) [inline src/assets/app.extensions.json:932]: just call it "reload: true"

File: src/app/components/toolbar/toggle-edit-offline/toggle-edit-offline.component.ts [hunk 3]
```diff
@@ -123,6 +125,11 @@ export class ToggleEditOfflineComponent implements OnInit {
     return this.alfrescoApiService.nodesApi.unlockNode(nodeId);
   }
 
+  private reloadOnlyOnViewerEditOfflineAction() {
```
Comment (DenysVuika) [inline src/app/components/toolbar/toggle-edit-offline/toggle-edit-offline.component.ts:128]: just call it "reload" maybe, and check the "this.data?.reload === true"

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: src/assets/app.extensions.json (hunk 2 above)
Comment (sivakumar414ram) [reply src/assets/app.extensions.json]: Updated

**Review by DenysVuika — APPROVED** [review]

**Comment by DenysVuika** [comment]
@sivakumar414ram there's one e2e is failing, probably needs to reload the list after the favorite actions

File: src/assets/app.extensions.json [hunk 4]
```diff
@@ -928,6 +928,9 @@
               "id": "app.viewer.toggleLock",
               "order": 100,
               "type": "custom",
+              "data": {
+                "reload": true
```
Comment (mauriziovitale) [inline src/assets/app.extensions.json:932]: do we use this strategy in other places?

**Comment by sivakumar414ram** [comment]
> @sivakumar414ram there's one e2e is failing, probably needs to reload the list after the favorite actions

Yup, some favorite e2e were failing, Was investigating and trying to use custom ToggleFavoriteComponent to toggle Favorites on the file preview mode now it works fine. Please review the changes again. Thanks.

**Comment by sivakumar414ram** [comment]
Closing this pr for now and this issue still in the discussion

**Pull request closed without merging**